### ISSUE TITLE
Presence of '\'' in the benchmark's name breaks the html report

### ISSUE THREAD

**Issue by jhrcek** [bos/criterion#202]
When I use apostrophe character in benchmark's name, all the charts in the html report are broken.

Steps to reproduce
1.  use apostrophe in benchmark's name:

```haskell
module Main where

import Criterion
import Criterion.Main

main :: IO ()
main = defaultMain
    [ env (return ()) $
       \ ~() -> bgroup "Jan's benchmarks" [bench "dummy" $ nf id ()]
    ]
```
2. compile the file and run it, enabling html report generation:

./mybench --output report.html

3. Open the resulting report in the browser (I use latest google-chrom version 69)

Actual result:
none of the charts is displayed, and browser's console shows this error
`Uncaught SyntaxError: Unexpected identifier`
Inspecting the report more closely the error is coming from this piece of code generated in the report based on benchmark's name:
```javascript
 var ylabels = [[-0,'<a href="#b0">Jan's benchmarks/dummy</a>'],];
```
Literal inclusion of benchmark's name into the javascript apparently leads to premature finish of the string literal, causing the sytax error.

**Comment by RyanGlScott** [comment]
Good catch. `criterion` uses the `microstache` library (a lighter version of `stache`) to substitute into `criterion`'s HTML template, and while `microstache` does correctly escape some characters, it seems to miss single quotes for some reason.

While #203 does fix this issue, I think a better fix would be to fix the upstream bug at `microstache`. I've submitted a bug report upstream at https://github.com/stackbuilders/stache/issues/35, and once that's fixed, I'll work on getting the fix into `microstache` itself.

**Comment by RyanGlScott** [comment]
Fixed in #203.
